tests/test_docker_wrapper.py

```diff
@@ -42,17 +44,70 @@ def test_json_input_with_format_succeeds(tmp_path):
         f.write(f'JSON{SEPARATOR}'.encode())
         f.write(json.dumps(INPUT_PARAMETERS).encode())
 
-    run_docker_wrapper(tmp_path, input_file)
+    check_wrapper_echoes_db(tmp_path, input_file)
 
 
-def run_docker_wrapper(tmp_path, input_file):
+def test_pickle_input_with_format_succeeds(tmp_path):
+    input_file = create_pickle_input(tmp_path)
+
+    check_wrapper_echoes_db(tmp_path, input_file)
+
+
+def test_wrapper_serializes_pickle_output(tmp_path):
+    input_parameters = {'method': 'hello_world', 'output_format': PICKLE}
+    input_file = create_pickle_input(tmp_path, input_parameters)
+
+    output_file = run_docker_wrapper(input_file, tmp_path)
+
+    with output_file.open('rb') as f:
+        assert f.read(len(PICKLE) + 1).decode() == f'{PICKLE}.'
+
+        result = pickle.loads(f.read())
+        pd.testing.assert_frame_equal(SAMPLE_DB, result)
+
+
+def test_wrapper_serializes_json_output(tmp_path):
+    input_parameters = {'method': 'hello_world', 'output_format': JSON}
+    input_file = create_pickle_input(tmp_path, input_parameters)
+
+    output_file = run_docker_wrapper(input_file, tmp_path)
+
+    with output_file.open('rb') as f:
+        assert f.read(len(JSON) + 1).decode() == f'{JSON}.'
```

Comment:
This seems strange to me that you have to read this in like this. It should also be clear from the extension that it is .json ? I have in my experience never seen a header in json files, but maybe I missed something?

Reply:
It's kind of a hack really. The reasoning behind this approach is that the input and output data is stored in a single field in the database as base64 encoded bytes. It could be in a separate field, but then you will have to worry about backwards compatibility. This way you don't have to migrate the database and the solution is still backwards compatible. You won't know from the extension what data format it is because the vantage6 node just loads the bytes from the database and puts it in a generic file. It doesn't know the data format either.

Reply:
Ok, would be nice to document this.

Reply:
It is documented in the next PR